libraries/incubator/ORM/Entity/EntityBuilder.php

```diff
@@ -280,7 +272,6 @@ public function handleHasMany(HasMany $relation, LocatorInterface $locator)
 			// Records from {$relation->entity} with {$relation->reference}={$id}
 			$repository = new Repository($relation->entity, $locator);
 			$entities   = $repository->findAll()->with($relation->reference, Operator::EQUAL, $id)->get();
-
 			$this->reflector->addField(new Field([
 				'name'  => $basename,
 				'type'  => 'relationData',
```

Comment:
Since clone() does not do a deep clone, caching the prototypes is not that easy. Needs a sophisticated __clone() method in Entity. Until then, this works fine.